torchrl/envs/utils.py

```diff
@@ -181,34 +187,46 @@ def step_mdp(
             next_tensordict.update(out)
             return next_tensordict
         return out
-    out = tensordict.get("next").clone(False)
-    excluded = set()
-    if exclude_done:
-        excluded.add(done_key)
-    if exclude_reward:
-        excluded.add(reward_key)
-    if len(excluded):
-        out = out.exclude(*excluded, inplace=True)
-    td_keys = None
+
+    td = tensordict.exclude("next")
+    td_next = tensordict.get("next")
+
+    td_keys = td.keys(True, True)
+    td_next_keys = td_next.keys(True, True)
```

Comment:
These are is the only time we traverse.
Basically we just visit every key in tensordict